engines/python/setup/djl_python/outputs.py

```diff
@@ -150,13 +154,43 @@ def encode(self) -> bytearray:
             self.write_utf8(msg, k)
             self.write_utf8(msg, v)
 
-        size = self.content.size()
-        msg += struct.pack('>h', size)
-        for i in range(size):
-            k = self.content.key_at(i)
-            v = self.content.value_at(i)
-            self.write_utf8(msg, k)
-            msg += struct.pack('>i', len(v))
-            msg += v
-
-        return msg
+        if self.stream_content is None:
```

Comment:
could we isolate this encoding component in to a single class? Maybe in the future it can be extended to support different encoding types